jcl/src/java.base/share/classes/java/security/AccessControlContext.java

```diff
@@ -331,22 +331,40 @@ public AccessControlContext(ProtectionDomain[] fromContext) {
 }
 
 AccessControlContext(AccessControlContext acc, ProtectionDomain[] context, int authorizeState) {
+	AccessControlContext(context, null, acc, authorizeState);
+}
+
+AccessControlContext(ProtectionDomain[] pdArray, AccessControlContext parentAcc, AccessControlContext context, int authorizeState) {
 	super();
 	switch (authorizeState) {
 	default:
 		// authorizeState can't be STATE_UNKNOWN, callerPD always is NULL
 		throw new IllegalArgumentException();
 	case STATE_AUTHORIZED:
-		if (null != acc) {
-			// inherit the domain combiner when authorized
-			this.domainCombiner = acc.domainCombiner;
+		if (null != context) {
+			if (parentAcc == null) {
+				// inherit the domain combiner when authorized
+				this.domainCombiner = acc.domainCombiner;
+			} else {
+				// when parent combiner is not null, use parent combiner to combine the current context
+				if (parentAcc.getCombiner() != null) {
+					this.context = parentAcc.getCombiner().combine(pdArray, context.context);
+					this.domainCombiner = parentAcc.getCombiner();
+				} else {
+					this.context = combinePDObjs(pdArray, context.context);
+					this.domainCombiner = context.domainCombiner;
+				}
+			}
+		} else {
+			this.domainCombiner = parentAcc.domainCombiner;
+			this.context = pdArray;
+			this.nextStackAcc = parentAcc;
```

Comment:
This changed the `AccessControlContext(AccessControlContext acc, ProtectionDomain[] context, int authorizeState)` behaviour.
To avoid the uncertainty of existing `AccessControlContext()` constructors which is being disabled permanently for JDK24+, I revert the previous suggestion. Pls separate this new constructor from others.